hermes-mock/src/main/java/pl/allegro/tech/hermes/mock/HermesMockDefine.java

```diff
@@ -40,14 +44,21 @@ public void avroTopic(String topicName, Response response) {
         addTopic(topicName, response, AVRO_BINARY);
     }
 
-    public <T> void avroTopic(String topicName, Response response, Schema schema, Class<T> clazz, Predicate<T> predicate){
-        AvroMatchesPattern<T> avroMatchesPattern = new AvroMatchesPattern<>(predicate, schema, clazz);
+    public <T> void avroTopic(String topicName, Response response, Schema schema, Class<T> clazz, Predicate<T> predicate) {
+        ValueMatcher<Request> avroMatchesPattern = ContentMatchers.matchAvro(hermesMockHelper, predicate, schema, clazz);
         addTopic(topicName, response, avroMatchesPattern);
     }
+
+    public <T> void jsonTopic(String topicName, Response response, Class<T> clazz, Predicate<T> predicate) {
+        ValueMatcher<Request> jsonMatchesPattern = ContentMatchers.matchJson(hermesMockHelper, predicate, clazz);
+        addTopic(topicName, response, jsonMatchesPattern);
+    }
+
     private void addTopic(String topicName, Response response, String contentType) {
         hermesMockHelper.addStub(topicName, response, contentType);
     }
-    private void addTopic(String topicName, Response response, AvroMatchesPattern avroMatchesPattern) {
-        hermesMockHelper.addStub(topicName, response, AVRO_BINARY, avroMatchesPattern);
+
+    private void addTopic(String topicName, Response response, ValueMatcher<Request> valueMatcher) {
+        hermesMockHelper.addStub(topicName, response, AVRO_BINARY, valueMatcher);
```

Comment:
As you're calling this method both in `jsonTopic` and `avroTopic` please take `contentType` as argument, and pass it instead of fixed `AVRO_BINARY`

Reply:
I didn't fail because while publishing a message I did:
```groovy
    private def publishJson(String topic, String message) {
        publisher.publish(topic, message.bytes)
    }
```
And that would be calling this method that would consume `AVRO_BINARY`:
```java
    @POST
    @Consumes(AVRO_BINARY)
    @Produces(APPLICATION_JSON)
    @Path("/{topicName}")
    Response publishMessage(@PathParam("topicName") String qualifiedTopicName, byte[] message);
```
I Fixed that